driver-core/src/main/java/com/datastax/driver/core/ArrayBackedResultSet.java

```diff
@@ -78,9 +80,14 @@ static ArrayBackedResultSet fromMessage(Responses.Result msg, SessionManager ses
                 // this explicitly because MultiPage implementation don't support info == null.
                 assert r.metadata.pagingState == null || info != null;
 
+                // If session is null, this is an internal call so we know it won't use custom types. It doesn't matter which registry is used.
+                CodecRegistry codecRegistry = (session == null)
```

Comment:
After JAVA-996, this can be replaced with `columnDefs.codecRegistry` - it is never null.